### DISCUSSION TITLE
Consider changing the proof verify as EigenLayer

### DISCUSSION THREAD

**Discussion by andreivladbrg** [sablier-labs/v2-periphery#336] [General]
Ref https://github.com/cantinasec/review-sablier-3/issues/3

**Comment by smol-ninja** [comment]
To add 

## Cons

1. It changes the API from 
```solidity
function verify(bytes32[] memory proof, bytes32 root, bytes32 leaf) { }
// to 
function verify(bytes32 memory proof, bytes32 root, bytes32 leaf, uint256 index) { }
```
2. More expensive than OZ's implementation.
3. Audited by Consensys

## Pros
1. Verify that all indexes are unique in the Merkle tree.
2. Ensure that proof length is valid.

Useful link:
- https://github.com/Layr-Labs/eigenlayer-contracts/blob/7229f2b426b6f2a24c7795b1a4687a010eac8ef2/src/contracts/libraries/Merkle.sol#L48-L78

@andreivladbrg I am happy to close it since we are not going to take any action at the moment and can re-open it in the future when we want to implement it.

**Reply by andreivladbrg** [reply]
yeah, we can close this
